src/Native/System.Native/pal_process.cpp

```diff
@@ -17,6 +17,12 @@
 #include <syslog.h>
 #include <unistd.h>
 
+#if HAVE_LIBPROC
+#include <libproc.h>
+#include <sys/proc_info.h>
+#include <sys/resource.h>
+#endif
```

Comment:
@jasonwilliams200OK, our current focus for the initial release is on Windows, Linux, and OS X.  Given that, and given that we have a working implementation on OS X, it's not a priority at present to spend time rewriting the existing working OS X-specific implementation.  If you or another member of the community would like to provide a sysctl-based FreeBSD implementation, we'd happily review the PR, and if that implementation also provided as good or better support for OS X as does the current OS X implementation, I think we'd happily get rid of the current implementation and switch to the new one.  I wouldn't expect such a PR to come from someone on the corefx team at Microsoft in the near future, though.  Thanks for your continued interest.